src/board.rs

```
@@ -142,6 +142,136 @@ impl Chessboard {
+            match piece {
+                'p' => {
+                    self.black_pawns &= clear_old; // Clear old position
+                    self.black_pawns |= add_new; // Set new position
+                }
+                'r' => {
+                    self.black_rooks &= clear_old;
+                    self.black_rooks |= add_new;
+                }
+                'b' => {
+                    self.black_bishops &= clear_old;
+                    self.black_bishops |= add_new;
+                }
+                'k' => {
+                    self.black_king &= clear_old;
+                    self.black_king |= add_new;
+                }
+                'q' => {
+                    self.black_queen &= clear_old;
+                    self.black_queen |= add_new;
+                }
+                'n' => {
+                    self.black_knights &= clear_old;
+                    self.black_knights |= add_new;
+                }
+                'P' => {
+                    self.white_pawns &= clear_old;
+                    self.white_pawns |= add_new;
+                }
+                'R' => {
+                    self.white_rooks &= clear_old;
+                    self.white_rooks |= add_new;
+                }
+                'B' => {
+                    self.white_bishops &= clear_old;
+                    self.white_bishops |= add_new;
+                }
+                'K' => {
+                    self.white_king &= clear_old;
+                    self.white_king |= add_new;
+                }
+                'Q' => {
+                    self.white_queen &= clear_old;
+                    self.white_queen |= add_new;
+                }
+                'N' => {
+                    self.white_knights &= clear_old;
+                    self.white_knights |= add_new;
+                }
+                _ => {}
+            }
+        } else {
+            println!("Invalid input position or piece");
+        }
```

We should have this panic instead of print.